src/main/scala/strawman/collection/View.scala

```diff
@@ -26,19 +26,20 @@ trait View[+A] extends Iterable[A] with IterableOps[A, View, View[A]] {
 /** This object reifies operations on views as case classes */
 object View extends IterableFactory[View] {
 
-  def fromIterator[A](it: => Iterator[A]): View[A] = new View[A] {
-    def iterator() = it
+  def fromIteratorProvider[A](it: () => Iterator[A]): View[A] = new View[A] {
+    def iterator() = it()
   }
 
   /** Avoid copying if source collection is already a view. */
-  def fromIterable[E](it: Iterable[E]): View[E] = it match {
-    case it: View[E] => it
-    case _ => View.fromIterator(it.iterator())
+  def from[E](it: IterableOnce[E]): View[E] = it match {
+    case it: View[E]     => it
+    case it: Iterable[E] => View.fromIteratorProvider(() => it.iterator())
+    case _ => scala.sys.error("One should not build a View from an IterableOnce instance")
```

Comment:
The problem that this PR _creates_ is that since `View[A]` extends `Iterable[A]`, it has to have an accompanying `IterableFactory[View]`, which itself has to provide an implementation of `from[E](it: IterableOnce[E]): View[E]`. However, we don’t really want to support building a `View` from an `Iterable`, because the resulting `View` would be effectively iterable only once. Hence, in such a case I throw an exception.

Alternatively, we could just silently copy the `IterableOnce` into an `ArrayBuffer` and call `.view` on it. But I’m not sure it’s better than throwing an exception because we want that building a `View` remains a cheap operation that doesn’t evaluate the underlying collection elements, right?

Reply:
This should fail at compile-time if at all possible.  I would rather silently copy than throw an exception if compile-time failure is not possible.

Reply:
I wasn't able to find a way to make it fail at compile time if View is an Iterable. The only solution that could work would be to make View something in between IterableOnce and Iterable.

Reply:
Can we
```scala
  type From[X] <: IterableOnce[X]
  from[E](it: From[E]): CC
```
in the factory, and then `type From[X] = IterableOnce[X]` in everything that's going to make a copy anyway and `type From[X] = Iterable[X]` in everything that refuses to make a copy?

Or does this propagate through other usages too far?

Reply:
This propagates quite a lot but maybe it’s not such a big deal actually… I’ll try again.

Reply:
No actually your trick worked fine with few changes. I just pushed a commit with that change.

Reply:
I updated the description of the PR with last changes (the introduction of the `IterableFactoryLike` generalization).